### PR TITLE
footer

### PR DESCRIPTION
**Pull request by roman89ua**
full integration ngx-translate in all modules,
all words from html of all project that was available are translated in vocabulary ,
translated login for, not found page, subject - not fully, navbar.
need pull.
ng buil --prop - ok   
https://i.imgur.com/fEwRdOU.png

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -19,6 +22,13 @@ import { getMatPaginatorUkr } from './shared/mat-paginator-config/mat-pagination
     BrowserModule,
     AppRoutingModule,
     HttpClientModule,
+    TranslateModule.forRoot({
+      loader: {
+        provide: TranslateLoader,
+        useFactory: HttpLoaderFactory,
+        deps: [HttpClient]
+      }
+    }),
```
Comment (yurakhomitsky) [inline src/app/app.module.ts:31]: Have you thought about import this module into the Shared and export?

File: src/app/app.module.ts [hunk 2]
```diff
@@ -33,3 +43,7 @@ import { getMatPaginatorUkr } from './shared/mat-paginator-config/mat-pagination
 })
 export class AppModule {
 }
+
+export function HttpLoaderFactory(http: HttpClient) {
+  return new TranslateHttpLoader(http);
+}
```
Comment (yurakhomitsky) [inline src/app/app.module.ts:49]: Can we create something like HttpLoader.ts  and export  this function? 
And in the TranslateModule use HttpLoaderFactory()

File: src/app/app.module.ts (hunk 1 above)
Comment (roman89ua) [reply src/app/app.module.ts:31]: there is only one TranslateModule.forRoot({
      loader: {
        provide: TranslateLoader,
        useFactory: HttpLoaderFactory,
        deps: [HttpClient]
      }
    }), 
in the project, why i shuld shered it?

File: src/app/app.module.ts (hunk 2 above)
Comment (roman89ua) [reply src/app/app.module.ts:49]: here you are probably right however first I want to make it work and then try to refactor this code when I see things in common

**Review by tatkaif — APPROVED** [review]

**Review by tatkaif — APPROVED** [review]

File: src/app/app.module.ts (hunk 2 above)
Comment (yurakhomitsky) [reply src/app/app.module.ts:49]: Okay)

File: src/app/app.module.ts (hunk 1 above)
Comment (yurakhomitsky) [reply src/app/app.module.ts:31]: You declared TranslateModule in each module, i think we can declare only in one module.

Comment (roman89ua) [reply src/app/app.module.ts:31]: I declared TranslateModule forRoot only in app module. other modules has TranslateModule forcChild

**Pull request merged**